apps/web/src/app/(main)/onboarding/form/create-form-dialog.tsx

```diff
@@ -0,0 +1,169 @@
+'use client';
+
+import React, { useState, useTransition } from 'react';
+import { useRouter } from 'next/navigation';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { InfoCircledIcon } from '@radix-ui/react-icons';
+import { useForm } from 'react-hook-form';
+import { toast } from 'sonner';
+import { z } from 'zod';
+
+import { Button } from '@formbase/ui/primitives/button';
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@formbase/ui/primitives/dialog';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@formbase/ui/primitives/form';
+import { Input } from '@formbase/ui/primitives/input';
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from '@formbase/ui/primitives/tooltip';
+
+import { api } from '~/lib/trpc/react';
+
+const FormSchema = z.object({
+  name: z.string().min(1, {
+    message: 'Form title is required.',
+  }),
+  returnUrl: z.string().optional(),
+  description: z.string().optional(),
+  keys: z.array(z.string()).optional(),
+});
+
+export function CreateFormDialog() {
+  const router = useRouter();
+  const form = useForm<z.infer<typeof FormSchema>>({
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      name: '',
+      description: '',
+      returnUrl: '',
+      keys: [''],
+    },
+  });
+  const [showDialog, setShowDialog] = useState<boolean>(false);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [_, startCreateTransaction] = useTransition();
+
+  const { mutateAsync: createNewForm } =
+    api.form.createOnboardingForm.useMutation();
+
+  const createPost = (data: z.infer<typeof FormSchema>) => {
+    startCreateTransaction(async () => {
+      await createNewForm(
+        {
+          title: data.name,
+          description: data.description,
+          returnUrl: data.returnUrl,
+        },
+        {
+          onSuccess: () => {
+            toast.success('New form endpoint created');
+            router.refresh();
+          },
+          onError: () => {
+            toast.error('Failed to create form, try again');
+          },
+        },
+      );
+    });
+  };
```

Comment:
Can we use `mutate` over `mutateAsync` so that we can get rid of the transitions we are using here? That will be a much  smaller code than this.

Reply:
the transition give the loading state, to disable the button. using mutate, we'll have to have a state to have a loading state and a set it true or false which is must easier with transitions